resources/metadata/transportRequestUploadSOLMAN.yaml

```diff
@@ -0,0 +1,92 @@
+metadata:
+  name: transportRequestUploadSOLMAN
+  aliases:
+    - name: transportRequestUploadFile
+  description: "Uploads content to a transport request"
+  longDescription: |
+    Uploads content to a transport request which is associated with a change document in SAP Solution Manager
+spec:
+  inputs:
+    secrets:
+      - name: uploadCredentialsId
```

Comment:
As far as I understood, uploadCredentialsId refers to the ABAP endpoint. so how is the id mapped to keys? which are the names of the keys. I found  yaml with the following entry:

```
 - name: username
        type: string
        ...
        secret: true
        resourceRef:
          - name: gitHttpsCredentialsId
            type: secret
            param: username
```

Dont we need such entry too?

Reply:
I don't know all the details here. Up to now I dont need it. In case we find out later that we in fact need it, it can be added (e.g. when we implement the "real" move from groovy to go. Up to now we add "only" the steps in go, but the corresponding coding in groovy is not yet removed. Before we can do that we have get feature complete (... traverse git commit history).

Reply:
The `username `corresponds to the OS user, `uploadCredentialsId `to the ABAP user. In case of a plain node container, I guess we have to pass the ABAP user, right? But the yaml doesnt contain it.

And isn't the OS user passed as ABAP user? see [transportRequestUploadSOLMAN.go](https://github.com/SAP/jenkins-library/blob/b2837e21cf53bfa6fe9d214d778ee41653fdb564/cmd/transportRequestUploadSOLMAN.go#L65)

Reply:
operating system user vs user which is used for the deployment is only important for cts upload. Here for solman where we use the `cm_client` the operating system user is not important at all. We can basically work with any operating system user.

We don't have a plain node container here, the step is exepected to be executed in conjunction with `ppiper/cm-client`.

Reply:
So i guess the user credentials comes in with `uploadCredentialsId`, right?

Reply:
> I guess the user credentials comes in with uploadCredentialsId, right?

Don't know all the details here. That will be important when we use the `piperExecuteBin` proxy in groovy. I would like to suggest to handle this at that later point in time.

> So there will be another PR which integrates it into the jenkins/groovy environment [...]

yes.